lib/echo/echo_client.rb

```diff
@@ -34,10 +34,39 @@ def get_phones(user_id, token)
       get("/echo-rest/users/#{user_id}/phones.json", {}, token_header(token))
     end
 
-    def get_preferences(user_id, token)
+    def general_contact_valid?(contact)
+      return false if contact.blank? || contact['phones'].blank? || contact['address'].blank?
+      true
+    end
+
+    def get_preferences(user_id, token, client, access_token)
       response = get("/echo-rest/users/#{user_id}/preferences.json", {}, token_header(token))
       if response.status == 404
         response = update_preferences(user_id, {preferences: {}}, token)
+      elsif response.status == 200 && !general_contact_valid?(response.body['preferences']['general_contact'])
```

Comment:
There are three "types" of users:
1. Users who have full profiles in ECHO (and in URS). These users exist before URS (with address, phone, and fax etc.)
2. Users who have partial profiles in ECHO - no address and/or phones. These users won't be able to successfully submit orders without a place holder in address and phone.
3. Users who have no profiles in ECHO. These users can't submit orders either.